xbmc/pvr/PVRDatabase.cpp

```diff
@@ -636,6 +636,18 @@ bool CPVRDatabase::Persist(CPVRChannelGroup &group)
   {
     CSingleLock lock(group.m_critSection);
 
+    /* Prevent AutoIncValue usage for internal groups as long they have fixed value */
+    if (group.GroupID() <= 0 && group.IsInternalGroup())
+      group.m_iGroupId = group.IsRadio() ?PVR_INTERNAL_GROUP_ID_RADIO:PVR_INTERNAL_GROUP_ID_TV;
+
+    /* First insert the # internal groups */
+    if(m_pDS->lastinsertid()==0)
+    {
+      strQuery = PrepareSQL("INSERT INTO channelgroups (idGroup) VALUES (NULL)");
+      ExecuteQuery(strQuery); //PVR_INTERNAL_GROUP_ID_RADIO
+      ExecuteQuery(strQuery); //PVR_INTERNAL_GROUP_ID_TV
```

Comment:
after a second look, we should store the complete group if not exists. Therefore we should call `CPVRDatabase::Persist(CPVRChannelGroup &group)` after we have checked if the group exists.

```
void CPVRDatabase::Insert(CPVRChannelGroupInternal &group)
{
  // check if group with id exists
  if (!exists)
    Persist(group);
}
```